### DISCUSSION TITLE
Mac Catalyst - How to inspect elements?

### DISCUSSION THREAD

**Discussion by rruprai1** [unoplatform/uno#18041] [General]
I've successfully gotten a webview2 to appear and load content in a Uno mac catalyst build. I'm trying next figure out how to attach the Safari web development tools to inspect my elements, look at JS console logs etc. 

However I can't find any good documentation on UNO to set this up. Is there a link or is this not possible?

**Comment by morning4coffe-dev** [comment]
Hey @rruprai1,

since I don't work with Mac, I may not be the best one to answer this question, but did you try following this doc https://platform.uno/docs/articles/uno-development/debugging-inspect-visual-tree.html? There is a section for most of our platforms and also a Retrieving the visual tree section. Is this what you are looking for?

**Reply by rruprai1** [reply]
Mostly yes, for macos, the iOS should hold true also, 
https://platform.uno/docs/articles/uno-development/debugging-inspect-visual-tree.html#ios

When I launch Safari to try and find WebView2 element to inspect, Safari can't find the app.

WebKit has the support:
https://webkit.org/blog/13936/enabling-the-inspection-of-web-content-in-apps/

Maui has a _way_, but is not currently applicable to UNO
https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/webview?view=net-maui-8.0&pivots=devices-maccatalyst#configure-the-native-webview-on-ios-and-mac-catalyst


Really I just need to figure out how the element can be made "inspectable" to allow web dev tools to debug the window, what I'm trying to do is see the JS console logs of the page being loaded.

**Reply by morning4coffe-dev** [reply]
I see, thanks for the clarification @rruprai1. My answer would be a bit vague on this topic, but one of my colleagues @ajpinedam, @MartinZikmund or @spouliot will be able to provide more details.

**Reply by ajpinedam** [reply]
Since iOS 16.4 it's required to turn on this feature as it's now off by default. We were not exposing it from Uno, but I just pushed a fix for it PR https://github.com/unoplatform/uno/pull/18074

Once the PR is merged, to enable Inspecting, you will need to enable the `IsInspectable` FeatureFlag by adding this line of code to your `App.xaml.cs`

```csharp
public App()
{
    this.InitializeComponent();
#if __IOS__
    Uno.UI.FeatureConfiguration.WebView2.IsInspectable = true;
#endif
}
```

**Reply by rruprai1** [reply]
Thanks. I'll try a dev branch when that PR is picked up. 

When would be the next big release for UNO, I know there are a number of bug fixes going in that I'm interested in.

**Reply by jeromelaban** [reply]
The next release should be soon, but this particular change may not be in that release. You will be able to test dev builds of the uno.sdk.

**Reply by rruprai1** [reply]
Does the new feature flag apply to Mac Catalyst? I am also trying to run on MacOS, open a webview2 and inspect.

**Reply by ajpinedam** [reply]
@rruprai1, the new flag will apply to both iOS and MacCatalyst.

**Reply by rruprai1** [reply]
I'm unable to see this work with a mac cataylst build (don't have iOS) 
Have set:
Uno.UI.FeatureConfiguration.WebView2.IsInspectable = true;

I have a simple page that opens a webview. Safari webtools still cannot picket up the webview for inspection.

**Reply by MartinZikmund** [reply]
@rruprai1 there was an additional change needed, should be available in upcoming releases

**Reply by rruprai1** [reply]
Thanks, is there a PR to follow?

**Reply by MartinZikmund** [reply]
It was merged into latest -dev releases already, https://github.com/unoplatform/uno/pull/18209 tracks backport to 5.4. Do you also need this in earlier version or are you ok with dev releases?

**Reply by rruprai1** [reply]
If the change is in the general 5.4 I would be happy :-) I can use the latest dev to try it out, which I think is 5.5.0-dev.23?

**Reply by rruprai1** [reply]
Looks like 5.5.0-dev.23 works and I can inspect the web contents using Safari Developement tools. 

Should be back ported to 5.4 release :-)

**Comment by rruprai1** [comment]
Issue resolved in 5.5.0-dev23. Should verify in 5.4 when available.
